src/core/lib/iomgr/tcp_client_posix.cc

```diff
@@ -336,6 +336,7 @@ int64_t grpc_tcp_client_create_from_prepared_fd(
     err = connect(fd, reinterpret_cast<const grpc_sockaddr*>(addr->addr),
                   addr->len);
   } while (err < 0 && errno == EINTR);
+  int saved_errno = (err < 0) ? errno : 0;
```

Comment:
nit: I suggest calling this `connect_errno`.

Reply:
I agree that's a better name. I used `saved_errno` to parallel the same logic in the event engine: https://github.com/grpc/grpc/blob/85cede5967eefd92703890b236939fa845d3801b/src/core/lib/event_engine/posix_engine/posix_engine.cc#L253-L262

Reply:
I noticed ... I'd love to see both variable names changed :)

Reply:
:)

I've now changed this one. I could send another PR for event_engine. (I think it would also be worthwhile in event_engine to fix a theoretical bug by adding the `(err < 0) ? errno : 0` condition I've written here; `errno` is undefined if the call succeeds.)

Reply:
I'd gladly review another fix PR! Thanks for this, I'll move forward with testing & landing it.